File: MyMindNotes/app/src/main/java/com/android/mymindnotes/AlarmBootReceiver.java
package com.android.mymindnotes;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.Log;

import java.util.Calendar;

public class AlarmBootReceiver extends BroadcastReceiver {
    SharedPreferences timeSave;

    @Override
    public void onReceive(Context context, Intent intent) {
        timeSave = context.getSharedPreferences("time", Activity.MODE_PRIVATE);

        // 부팅 시 알람 설정을 위한 코드
        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
            Log.e("MyChecker:BootCheck", "BootCheck : " + "여기까지 무사히 들어왔다.");
            // Set the alarm here.
            if (timeSave.getLong("time", 0) != 0) {
                // 저장된 알람이 있을 경우에만 reset alarm
                Calendar calendar = Calendar.getInstance();
                calendar.setTimeInMillis(timeSave.getLong("time", 0));
                AlarmSetting.setAlarm(calendar, context);
                Log.e("MyChecker:BootCheck", "BootCheck : " + calendar.getTime());
            }
        }
    }
}

File: MyMindNotes/app/src/main/java/com/android/mymindnotes/domain/AlarmBootReceiver.kt
package com.android.mymindnotes.domain

import android.content.BroadcastReceiver
import android.content.Intent
import android.content.Context
import android.os.Build
import android.util.Log
import androidx.annotation.RequiresApi
import com.android.mymindnotes.domain.usecase.AlarmUseCase
import com.android.mymindnotes.hilt.module.MainDispatcherCoroutineScope
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.launch
import java.util.*
import javax.inject.Inject

@AndroidEntryPoint
class AlarmBootReceiver: BroadcastReceiver(){

    @Inject
    lateinit var alarmUseCase: AlarmUseCase

    @MainDispatcherCoroutineScope
    @Inject
    lateinit var mainDispatcherCoroutineScope: CoroutineScope

    @RequiresApi(Build.VERSION_CODES.N)
    override fun onReceive(context: Context, intent: Intent) {

        // Code to set alarm on boot
        if (intent.action == "android.intent.action.BOOT_COMPLETED") {
            Log.e("알람 체크", "BootCheck : 부팅 액션까지 들어옴")
            // Set the alarm here.
            mainDispatcherCoroutineScope.launch {
                val rebootTime = alarmUseCase.getRebootTime().first()
                if (rebootTime != 0L) {
                    // reset alarm only if there is a saved alarm
                    val calendar = Calendar.getInstance()
                    calendar.timeInMillis = rebootTime
                    alarmUseCase.setAlarm(calendar, context)
                    Log.e("알람 체크", "BootCheck : " + calendar.time)
                }
            }
        }
    }
}